### PR TITLE
Define a preprocessor flag for enabling warnings on unused ConnectionHandle

### PR DESCRIPTION
**Pull request by phyBrackets**
Resolving #43

### CODE REVIEW

File: src/kdbindings/signal.h [hunk 1]
```diff
@@ -11,6 +11,11 @@
 
 #pragma once
 
+// Enables compiler warnings for unused ConnectionHandle results.
+#ifndef KDBINDINGS_WARN_UNUSED
+#define KDBINDINGS_WARN_UNUSED [[nodiscard]]
```
Comment (seanharmer) [inline src/kdbindings/signal.h:16]: May as well use [[nodiscard]] directly if just defining it.

Or if you want to make it optional we should make it an option in CMake.

To me it seems strange asking downstream to define KDBINDINGS_WARN_UNUSED if they *don't* want warnings. :)

Comment (phyBrackets) [reply src/kdbindings/signal.h]: We have to document this somewhere as well, right?

**Review by LeonMatthesKDAB — CHANGES_REQUESTED** [review]
Overall a good change.
The way it's currently implemented, it would make CMake required, which I would like to avoid.

That should easy to fix though.

**Review by phyBrackets — COMMENTED** [review]
Thanks @LeonMatthesKDAB , Looks good to me now!

**Comment by LeonMatthesKDAB** [comment]
@phyBrackets cool, can you please incorporate my last review, so we can merge the PR?

File: cmake/KDBindingsConfig.h.in [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+// Check if KDBINDINGS_ENABLE_WARN_UNUSED option is set
+#cmakedefine KDBINDINGS_ENABLE_WARN_UNUSED
+
+#ifdef KDBINDINGS_ENABLE_WARN_UNUSED
+#define KDBINDINGS_WARN_UNUSED [[nodiscard]]
+#else
+#define KDBINDINGS_WARN_UNUSED
+#endif
```
[the remark this answers is not in the document]
Comment (phyBrackets) [reply cmake/KDBindingsConfig.h.in]: I am not sure but `add_compile_definitions` doesn't seems to work as expected, I have used it but `KDBINDINGS_ENABLE_WARN_UNUSED` stay disable during the use of it in the preprocessor definitions header file, while `target_compile_definitions` works correctly.

**Comment by LeonMatthesKDAB** [comment]
@phyBrackets Thanks for the update, the code is good now :+1: 

It just seems we have loads of unused ConnectionHandles in our tests :thinking: 
I guess we could use [std::ignore](https://en.cppreference.com/w/cpp/utility/tuple/ignore) to fix this.

It does bring up the question of how annoying this would be for end-users though.
@seanharmer If we enable this by default, do you think e.g. Serenity would explode with warnings about unused ConnectionHandles?
Because then I may need to revert my opinion about making this ON by default.

If it's just our testing that discards connection handles that often, I'm fine with leaving it ON though.

**Comment by seanharmer** [comment]
I'd suggest enabling it by default. It usually indicates something we're doing wrong so good to know and fix them.

**Comment by LeonMatthesKDAB** [comment]
@seanharmer thanks for the feedback :+1: .

@phyBrackets can you please add the std::ignore calls where required, or use the ConnectionHandle if it makes sense, so CI passes.
Then we can merge the PR :partying_face: 

Side note: To test locally, you should be able to just run `cmake --preset=clazy && cmake --build --preset=clazy && ctest --preset=clazy`, which enables all warnings.

**Comment by phyBrackets** [comment]
> @phyBrackets Thanks for the update, the code is good now 👍
> 
> It just seems we have loads of unused ConnectionHandles in our tests 🤔 I guess we could use [std::ignore](https://en.cppreference.com/w/cpp/utility/tuple/ignore) to fix this.
> 
> It does bring up the question of how annoying this would be for end-users though. @seanharmer If we enable this by default, do you think our existing user code would explode with warnings about unused ConnectionHandles? Because then I may need to revert my opinion about making this ON by default.
> 
> If it's just our testing that discards connection handles that often, I'm fine with leaving it ON though.

Hi @LeonMatthesKDAB , Thanks for the idea. I guess it would be fine with `std::ignore` as well but it seems more intended to be used with tuples and it is also not a recommended way by standard to be used for suppressing warning in this case as it could potentially break in future version of the spec. Casting to simply `void` seems fine but I am not sure about if it looks good for our examples? (fine for tests!)

**Pull request merged**